src/inverted_index.c

```diff
@@ -329,19 +329,19 @@ ENCODER(encodeNumeric) {
     if (realVal == -INFINITY) {
       header.encFloat.sign = 1;
     }
-
   } else {
     // Floating point
     NumEncodingFloat *encFloat = &header.encFloat;
-    if (fabs(absVal - f32Num) < 0.01) {
+    if (realVal == f32Num) {
```

Comment:
You might as well remove the entire float code, if the number is not integral, since the chances that a non-integral float and a double will == the same is almost none.

Reply:
While true, we need to keep it in the decoder for back compatibility so let's keep it in the encoder as well.